qiskit_experiments/characterization/t2ramsey.py

```diff
@@ -99,3 +102,203 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+    """
+
+    __analysis_class__ = RamseyXYAnalysis
+
+    def __init__(
+            self,
+            qubit: int,
+            delays: Union[Iterable[float], np.ndarray],
+            unit: str = "s",
+            offset_frequency: float = 0.,
+    ):
+        """Initialize new experiment.
+
+        Args:
+            qubit: The qubit under test.
+            delays: Delay times of the experiments.
+            unit: Time unit of `delays`. One of 's', 'ms', 'us', 'ns', 'ps', 'dt'.
+            offset_frequency: Frequency offset for this experiment.
+
+        .. warning::
+
+            ``offset_frequency`` doesn't really change the qubit frequency for now,
+            because of the missing oscillator frequency upgrade pass in the circuit execution.
+        """
+        super().__init__([qubit])
+
+        default_options = self.experiment_options
+        default_options.delays = delays
+        default_options.unit = unit
+        default_options.offset_frequency = offset_frequency
+
+        self.set_experiment_options(**default_options.__dict__)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        return Options(
+            delays=None,
+            unit="s",
+            offset_frequency=0.,
+            calibrations={
+                "sx": None,
+                "measure": None,
+            },
+        )
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Return a list of experiment circuits.
+
+        Args:
+            backend: Optional, a backend object.
+
+        Returns:
+            A list of :class:`QuantumCircuit`.
+        """
+        options = self.experiment_options
+
+        p_delay = Parameter("delay")
+
+        ram_x = QuantumCircuit(1)
+        ram_x.sx(0)
+        ram_x.delay(p_delay, 0, unit=options.unit)
+        ram_x.sx(0)
+        ram_x.measure_active()
+
+        ram_y = QuantumCircuit(1)
+        ram_y.sx(0)
+        ram_y.delay(p_delay, 0, unit=options.unit)
+        ram_y.rz(-np.pi/2, 0)
+        ram_y.sx(0)
+        ram_y.measure_active()
+
+        circs = []
+        for delay in options.delays:
+
+            # format delay to SI unit for analysis
+            if options.unit == "dt":
+                xval = delay * backend.configuration().dt
+            else:
+                if options.unit == "s":
+                    xval = delay
+                else:
+                    xval = apply_prefix(delay, options.unit)
+
+            metadata = {
+                "experiment_type": self._type,
+                "qubits": (self.physical_qubits[0], ),
+                "delay": delay,
+                "unit": options.unit,
+                "xval": xval,
+            }
+
+            # create ramsey x
+            assigned_x = ram_x.assign_parameters({p_delay: delay}, inplace=False)
+            assigned_x.metadata = metadata.copy()
+            assigned_x.metadata["post_pulse"] = "x"
```

Comment:
maybe `measurement_axis`, or `axis`, or `meas_axis` instead of `post_pulse`?

Reply:
Note that meas-axis and post pulse is not the same label, for example, if we rotate about x (post pulse=x) the measure axis will be y. Actually I don't know exact definition of which one is Ramsey X (Y). In the class documentation, I define one with SX pulse as Ramsey X, but if this is defined based on measure axis, that should be called Ramsey Y. @oliverdial?

Reply:
No strong opinion here sorry -- pick one and be consistent?

Reply:
Thanks. If you don't have preference, we can define it based on measurement axis as @eggerdj says? Namely, sx-delay-sx := Ramsey Y and sx-delay-sy := Ramsey X. I'll define this in the class documentation.